app/recipe_blue_print/views.py

```diff
@@ -0,0 +1,198 @@
+            'ingredients': recipe.ingredients,
+            'date_created': recipe.date_created,
+            'date_modified': recipe.date_modified,
+            'created_by': recipe.created_by
+        })
+        return make_response(response), 200
+    else:
+        return make_response(jsonify({"message": "Page not Found"})), 404
+
+
+@swag_from('swagger_docs/search_current_user_recipes.yaml', methods=['GET'])
+def search_by_name():
+    user = get_authenticated_user(request)
+    if not user:
+        return make_response(jsonify({"message": "You have no access rights"})), 403
+
+    if request.method == "GET":
+        try:
+            # get params
+            q = request.args.get('q') or " "
+            limit = request.args.get('limit') or 20
+            page = request.args.get('page') or 1
+
+            q = str(q)
+            limit = int(limit)
+            page = int(page)
+
+            recipe_by_name = Recipe.query.filter_by(created_by=user.id).filter(Recipe.name.like('%' + q + '%'))\
+                .paginate(per_page=limit, page=page, error_out=False)
+            if not recipe_by_name:
+                abort(404)
+            obj = []
+            for recipe in recipe_by_name.items:
+                rec = {
+                    'id': recipe.id,
+                    'category_id': recipe.category_id,
+                    'name': recipe.name,
+                    'details': recipe.details,
+                    'ingredients': recipe.ingredients,
+                    'date_created': recipe.date_created,
+                    'date_modified': recipe.date_modified,
+                    'created_by': recipe.created_by,
+                    'per_page': recipe_by_name.per_page,
+                    'page_number': recipe_by_name.page,
+                    'total_items_returned': recipe_by_name.total
+                }
+                obj.append(rec)
+            if not obj:
+                return make_response(jsonify({
+                    'message': 'No Content On This Page or Search Not Found',
+                    'per_page': recipe_by_name.per_page,
+                    'page_number': recipe_by_name.page,
+                    'total_items_returned': recipe_by_name.total
+                }))
+            return make_response(jsonify(obj)), 200
+        except Exception:
+            return make_response(jsonify({'message': 'limit and page cannot be string values'})), 400
+
+# Define the rule for recipes url ---> /recipes or /recipes?limit=<int:limit>&page=<int:page>
+recipe_blue_print.add_url_rule('/recipes', view_func=recipes_view, methods=['POST', 'GET'])
+# Define the rule for recipes url ---> /recipes/<int:id>
+recipe_blue_print.add_url_rule('/recipes/<int:id>', view_func=recipes_view_edit, methods=['DELETE', 'PUT', 'GET'])
+# Define the rule for recipes url ---> /recipes/search?q=<sting:q>&limit=<int:limit>&page=<int:page>'
+recipe_blue_print.add_url_rule('/recipes/search', view_func=search_by_name, methods=['GET'])
+
```

Comment:
Make use of class based views i.e.  `MethodView` makes code much cleaner.
Look into how to cover the method for `/recipes/search` url under the method for `/recipes` url